action/register.go

```diff
@@ -0,0 +1,121 @@
+package action
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+
+	"github.com/allegro/vaas-registration-hook/k8s"
+	"github.com/allegro/vaas-registration-hook/vaas"
+)
+
+const (
+	// RegisterName is the CLI name of this action
+	RegisterName = "register"
+
+	// FlagWeight is the weight that a backend in VaaS is initially assigned
+	FlagWeight = "weight"
+	// FlagDC represents datacenter short name as defined in VaaS
+	FlagDC = "dc"
+
+	canaryTag = "canary"
+)
+
+// GetRegisterFlags returns a list of flags available for this action
+func GetRegisterFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.IntFlag{
+			Name:  FlagWeight,
+			Usage: "initial weight of this backend",
+			Value: 1,
+		},
+		cli.StringFlag{
+			Name:  FlagDC,
+			Usage: "datacenter short name as defined in VaaS",
+		},
+	}
+}
+
+// RegisterCLI configures a VaaS client from CLI data and runs register()
+func RegisterCLI(c *cli.Context) error {
+	config := getCommonParameters(c.Parent().Parent())
+	debug(config)
+
+	apiClient := vaas.NewClient(config.VaaSURL, config.VaaSUser, config.VaaSKey)
+	weight := c.Int(FlagWeight)
+	dcName := c.String(FlagDC)
+
+	return register(apiClient, config, weight, dcName)
+}
+
+// RegisterK8s configures a VaaS client from K8s data and runs register()
+func RegisterK8s(podInfo *k8s.PodInfo, config CommonConfig) error {
+	config.Address = podInfo.GetPodIP()
+	config.Port = podInfo.GetDefaultPort()
+	director, err := podInfo.GetDirector()
+	if err == nil {
+		config.Director = director
+	} else {
+		log.Errorf("could not find VaaS director in Pod info: %s", err)
```

Comment:
We could as well return wrapped error here, no need to continue if director is not known imo.